conans/client/generators/virtualenv_python.py

```diff
@@ -2,12 +2,20 @@
 
 
 class VirtualEnvPythonGenerator(VirtualRunEnvGenerator):
+
     def __init__(self, conanfile):
         super(VirtualEnvPythonGenerator, self).__init__(conanfile)
         self.venv_name = "conanenvpython"
-        self.env["PYTHONPATH"] = list(
-            set(
-                conanfile.env.get("PYTHONPATH", [])
-                + self.deps_env_info.vars.get("PYTHONPATH", [])
-            )
-        )
+        ppath = conanfile.env.get("PYTHONPATH")
```

Comment:
When a Conan package does not supply a `PYTHONPATH` the activate script ends up with a `"None"`: `PYTHONPATH="None"${PYTHONPATH+:$PYTHONPATH}`
One woule usually not use `virtualenv_python` when no packages defines a `PYTHONPATH`, but maybe it is better to use `conanfile.env.get("PYTHONPATH", [])`?

Reply:
Yes, good catch!